### PR TITLE
Deltann serving

### PR DESCRIPTION
**Pull request by itamaker**
#54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -217,7 +217,7 @@ Please cite this [paper](https://arxiv.org/abs/1908.01853) when referencing DELT
          {Gao}, Yonghu and {Guo}, Tingwei and {Zhang}, Yi and {He}, Yahao and
          {Ma}, Baochang and {Zhou}, Yulong and {Zhang}, Kangli and {Liu}, Chao and
          {Lyu}, Ying and {Wang}, Chenxi and {Gong}, Cheng and {Wang}, Yunbo and
-         {Zou}, Wei and {Song}, Hui and {Li}, Xiangang},
+         {Zou}, Wei and {Song}, Hui and {Li}, Xiangang and {Jia}, Zhaoyang},
```
Comment (zh794390558) [inline README.md]: remove this, since I will update later.

File: deltann/examples/text_cls/test.cc [hunk 2]
```diff
@@ -29,9 +29,9 @@ int main(int argc, char** argv) {
 
   int in_num = 1;
   Input ins[1];
-  const char* text = "I'm angry.";
+  const char* text = "懂了客服";
   ins[0].ptr = (void*)(&text);
-  ins[0].size = 1;
+  ins[0].size = sizeof(text);
```
Comment (zh794390558) [inline deltann/examples/text_cls/test.cc]: sizeof(text) + 1 for `/0`

File: deltann/server/core/delta_serving.go [hunk 3]
```diff
@@ -49,9 +53,14 @@ func DeltaListen(opts DeltaOptions) error {
...
 		return err
 	}
+
+	dispatcher = DeltaDispatcher(conf.DeltaConf.DeltaServingPoll.DeltaMaxWorker, conf.DeltaConf.DeltaServingPoll.DeltaMaxQueue)
+	dispatcher.Run()
+
 	glog.Infof("start deltaModelRun...")
 	router := gin.Default()
-
+	pprof.Register(router)
```
Comment (zh794390558) [inline deltann/server/core/delta_serving.go]: add one flag to control `pprof`

File: deltann/server/core/model/delta_model.go [hunk 4]
```diff
@@ -72,7 +81,7 @@ func DeltaModelRun(valueInputs interface{}) (string, error) {
 		deltaPtr := C.CString(valueInputs.(string))
 		defer C.free(unsafe.Pointer(deltaPtr))
 		ins.ptr = unsafe.Pointer(deltaPtr)
-		ins.size = C.int(len(valueInputs.(string)))
+		ins.size = C.int(len(valueInputs.(string)) + 1)
```
Comment (zh794390558) [inline deltann/server/core/model/delta_model.go]: putting `+1` outter.

File: deltann/server/examples/main.go [hunk 5]
```diff
@@ -22,10 +22,13 @@ import (
 )
 
 func main() {
+	deltaPort := flag.String("port", "none", "set http listen port")
+	deltaYaml := flag.String("yaml", "none", "set delta model yaml conf")
+	deltaType := flag.String("type", "none", "set server type")
```
Comment (zh794390558) [inline deltann/server/examples/main.go]: add more details of the deltaType for what.

File: deltann/server/core/model/delta_model.go (hunk 4 above)
Comment (itamaker) [reply deltann/server/core/model/delta_model.go]: this is unique to string and is already in the case string

**Pull request merged**